src/components/task-plan/builder/tasking-date-times.cjsx

```diff
@@ -16,20 +16,17 @@ TaskingDateTimes = React.createClass
   propTypes:
     id:                  React.PropTypes.string.isRequired
     isEditable:          React.PropTypes.bool.isRequired
-    taskingOpensAt:      React.PropTypes.string.isRequired
-    taskingDueAt:        React.PropTypes.string
-    setDueAt:            React.PropTypes.func.isRequired
-    setOpensAt:          React.PropTypes.func.isRequired
-    dueTime:             React.PropTypes.string
-    openTime:            React.PropTypes.string.isRequired
-    defaultDueTime:      React.PropTypes.string.isRequired
-    defaultOpenTime:     React.PropTypes.string.isRequired
     isVisibleToStudents: React.PropTypes.bool
     period:              React.PropTypes.object
 
-  isPastDue: (date, time) ->
-    return false unless date?
-    TimeHelper.makeMoment("#{date} #{time}").isBefore(TimeHelper.makeMoment())
+  getError: ->
+    return false unless @refs?.due?.hasValidInputs() and @refs?.open?.hasValidInputs()
+
+    {id, period} = @props
+
+    return false if TaskingStore.isTaskingValid(id, period)
+
+    _.first(TaskingStore.getTaskingErrors(id, period))
```

Comment:
the first one is the right one to display as you would not expect the user to try to address all errors at once.  errors could include misformatted date, misformatted time, and time range being wrong.